packages/modular-scripts/src/build/dependencyManifest.ts

```diff
@@ -0,0 +1,66 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { Project } from 'ts-morph';
+import type { CoreProperties } from '@schemastore/package';
+import getModularRoot from '../utils/getModularRoot';
+import getLocation from '../utils/getLocation';
+
+type DependencyManifest = Record<string, string>;
+
+const npmPackageMatcher =
+  /^(@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*/;
+
+// Get dependencies from import / require declarations, since they could be hoisted to the root workspace
+function getDependenciesFromSource(workspaceLocation: string) {
+  const project = new Project();
+  project.addSourceFilesAtPaths(
+    path.join(workspaceLocation, 'src/**/*{.d.ts,.ts,.js,.jsx,.tsx}'),
+  );
+  const dependencySet = new Set(
+    project.getSourceFiles().flatMap((sourceFile) =>
+      sourceFile
+        .getImportDeclarations()
+        .map(
+          (declaration) =>
+            declaration.getModuleSpecifierValue().match(npmPackageMatcher)?.[0],
+        )
+        .filter(Boolean),
+    ),
+  ) as Set<string>;
+
+  return Array.from(dependencySet);
+}
+
+export async function generateDependencyManifest(target: string) {
+  // This is based on the assumption that nested package are not supported, so dependencies can be either declared in the
+  // target's package.json or hoisted up to the workspace root.
+  const targetLocation = await getLocation(target);
+
+  const rootPackageJsonDependencies =
+    (
+      fs.readJSONSync(
+        path.join(getModularRoot(), 'package.json'),
+      ) as CoreProperties
+    ).dependencies || {};
+
+  const targetPackageJsonDependencies =
+    (
+      fs.readJSONSync(
+        path.join(targetLocation, 'package.json'),
+      ) as CoreProperties
+    ).dependencies || {};
+
+  const manifest = getDependenciesFromSource(
+    targetLocation,
+  ).reduce<DependencyManifest>((manifest, depName) => {
+    const depVersion =
```

Comment:
This would be cleaner by creating the dependencies by overwriting from the `targetLocation` into the root dependencies. This would create a "workspace" set of dependencies.

Reply:
Done

Reply:
@LukeSheard not sure I understand. Does this mean initialise the reducer with the target dependencies, then add the root dependencies as we iterate through them?